.github/actions/setup/action.yml

```diff
@@ -22,51 +22,179 @@ inputs:
   python-version:
     required: true
     type: string
+  cuda-version:
+    required: true
+    type: string
 
 runs:
   using: composite
   steps:
-    - name: Set REPO_DIR and Dump environment
+    # WAR: setup-python is not relocatable...
+    # see https://github.com/actions/setup-python/issues/871
+    - name: Set up Python ${{ inputs.python-version }}
+      if: ${{ startsWith(inputs.host-platform, 'linux') }}
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+
+    # - name: Set up new PowerShell (pwsh)
```

Comment:
My gut tells me we should have the windows stuff uncommented before approval, or take all windows related changes out of this review. Having the install-vs script included with this commented out for example. Per your recent comments it seems this is being address by the nv-gha-runnners issues. Do we want to consider those dependencies?

Reply:
nvm per plan to switch to nv pre installed runners. Will wait to approve until that stuff is removed?

Reply:
- Using GH-hosted runners: pwsh and msvc are both included in the Windows images.
- Using self-hosted (nv-gha) runners: Yes we need them to support both, this is currently a blocker tracked in both vm-images issues.